### PR TITLE
[13.0][MIG]pos_margin

### PR DESCRIPTION
**Pull request by manuelregidor**
Standard migration
Supersedes https://github.com/OCA/pos/pull/399

### CODE REVIEW

**Review by HaraldPanten — APPROVED** [review]

**Review by ValentinVinagre — APPROVED** [review]
LGTM 👍🏻

**Comment by HaraldPanten** [comment]
@legalsylvain I think this one can be merged. THX!

**Comment by ivantodorovich** [comment]
Can you please integrate the latest changes recently merged in 12.0?

**Comment by ivantodorovich** [comment]
Thanks for the reaction @manuelregidor 
I see that you included the latests changes I mentioned in your Migration to 13.0 commit.

In fact we want to cherry-pick the missing commits instead, in order to respect authors and to keep the proper commit history.

In this case, I believe these 3 commits are missing:

- https://github.com/OCA/pos/commit/d97f96c86e31a307c28066504fc6c99486ad46ad
- https://github.com/OCA/pos/commit/9bb3c18e5dfbfe148858a405a835ffdc0b0fc494
- https://github.com/OCA/pos/commit/3283582bd60c72ec170e913b4a818fd2bbe838f3

File: pos_margin/models/pos_order.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+class PosOrder(models.Model):
+    _inherit = "pos.order"
+
+    # Columns Section
+    margin = fields.Float(
+        string="Margin",
+        compute="_compute_margin",
+        store=True,
+        digits=dp.get_precision("Product Price"),
```
Comment (ivantodorovich) [inline pos_margin/models/pos_order.py]: Runbot complaining about:
`2021-07-16 10:09:40,403 160 WARNING openerp_test odoo.addons.base.models.decimal_precision: Deprecated call to decimal_precision.get_precision(<application>), use digits=<application> instead`

Check other lines like this, too

**Review by ivantodorovich — CHANGES_REQUESTED** [review]
Travis is complaining too, because of pre-commit.
Can you check? I believe it's ready to merge as soon as it's green

File: pos_margin/models/pos_order.py (hunk 1 above)
Comment (manuelregidor) [reply pos_margin/models/pos_order.py]: Done!

File: pos_margin/models/pos_order_line.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+class PosOrderLine(models.Model):
+    _inherit = "pos.order.line"
+
+    # Columns Section
+    margin = fields.Float(
+        string="Margin",
+        compute="_compute_multi_margin",
+        store=True,
+        digits=dp.get_precision("Product Price"),
```
Comment (ivantodorovich) [inline pos_margin/models/pos_order_line.py]:
```suggestion
        digits="Product Price",
```

File: pos_margin/models/pos_order_line.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    )
+
+    margin_percent = fields.Float(
+        string="Margin (%)",
+        compute="_compute_multi_margin",
+        store=True,
+        digits=dp.get_precision("Product Price"),
```
Comment (ivantodorovich) [inline pos_margin/models/pos_order_line.py]:
```suggestion
        digits="Product Price",
```

File: pos_margin/models/pos_order_line.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    )
+
+    purchase_price = fields.Float(
+        string="Cost Price",
+        compute="_compute_multi_margin",
+        store=True,
+        digits=dp.get_precision("Product Price"),
```
Comment (ivantodorovich) [inline pos_margin/models/pos_order_line.py]:
```suggestion
        digits="Product Price",
```

File: pos_margin/models/pos_order_line.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# Copyright (C) 2017 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+import odoo.addons.decimal_precision as dp
```
Comment (ivantodorovich) [inline pos_margin/models/pos_order_line.py]:
```suggestion
```

**Review by ivantodorovich — CHANGES_REQUESTED** [review]
Runbot's still complaining, there are more lines with `dp` :sweat_smile:

**Comment by manuelregidor** [comment]
@ivantodorovich Done! Thank you!

**Review by ivantodorovich — APPROVED** [review]

**Pull request merged**